http-client/Network/HTTP/Client/Types.hs

```diff
@@ -547,6 +548,18 @@ data Request = Request
     -- dealing with implicit global managers, such as in @Network.HTTP.Simple@
     --
     -- @since 0.4.28
+
+    , requestMaxAttempts :: Maybe Int
+    -- ^ The 'Request' will be retried if all the following conditions are met:
+    --
+    -- * this value is greater than 1 or @Nothing@,
+    -- * the exception is retryable as defined by 'managerRetryableException'.
+    --
+    -- Note: Exceptions thrown via 'checkResponse' will not trigger a retry.
+    --
+    -- Default: Just 1
+    --
+    -- @since
```

Comment:
Can you add a minor version bump, fill in this `@since`, and add a note to the ChangeLog?